metricbeat/docs/modules/ibmmq.asciidoc

```diff
@@ -0,0 +1,53 @@
+////
+This file is generated! See scripts/mage/docs_collector.go
+////
+
+[[metricbeat-module-ibmmq]]
+[role="xpack"]
+== IBM MQ module
+
+beta[]
+
+This module periodically fetches metrics from IBM MQ.
+
+[float]
+=== Compatibility
+
+The ibmmq `status` metricset is compatible with any IBM MQ version exposing metrics in Prometheus format.
+
+The official Docker container provides an option to easily enable metrics exporter using a dedicated environment
+variable:
+
+`MQ_ENABLE_METRICS` - Set this to `true` to generate Prometheus metrics for the Queue Manager.
+
```

Comment:
Do we know which versions of MQ provide the exporter? Do we know how to enable it when not using the Docker image? I'm thinking on how feasible / easy is to go this route in real world deployments (/cc @felix-lessoer @exekias )

Reply:
Thank you for the questions, Andres. Let me respond to these:

>  Do we know how to enable it when not using the Docker image?

The exporter is available only if you're using the image. It's part of the `runmqserver`, an application supported running in containerized environments.

The description of the internals is available here: https://github.com/ibm-messaging/mq-container/blob/master/docs/internals.md

Source (main.go) of `runmqserver`: https://github.com/ibm-messaging/mq-container/blob/master/cmd/runmqserver/main.go

> Do we know which versions of MQ provide the exporter?

Not specifically MQ, but the containerized distribution of it. According to the sources, the metrics exporter has been enabled in the release version 9.1.0.

Source code: https://github.com/ibm-messaging/mq-container/tree/9.1.0/internal/metrics

PR: https://github.com/ibm-messaging/mq-container/pull/81/files

> I'm thinking on how feasible / easy is to go this route in real world deployments

I think that we can leave a message here that Metricbeat only supports deployments of the official containerized distribution of the MQ. Currently it's the only way to go if we don't want to link the proprietary code.